app/models/person.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+# Person class
+class Person < ApplicationRecord
+  include Roleable
+
+  has_and_belongs_to_many :roles
+  has_and_belongs_to_many :movies
+
+  accepts_nested_attributes_for :people_roles
+
+  validates :first_name, :last_name, :aliases, presence: true
+  validates :genre,
+            inclusion: %w[male female],
+            allow_nil: false
+
+  def movies_as(role_name)
```

Comment:
I suggest to validate that person actually owns that role, using `has_role`, before making db query search. (this would save db time)